dask/dataframe/partitionquantiles.py

```diff
@@ -413,7 +414,7 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
     if is_categorical_dtype(data):
         data = data.codes
         interpolation = "nearest"
-    elif np.issubdtype(data.dtype, np.integer):
+    elif np.issubdtype(data.dtype, np.integer) and not is_cupy_type(data):
```

Comment:
Thanks @pentschev! Do you think it's worth adding a comment saying that "nearest" option is not supported for cupy here ?

Reply:
Done in https://github.com/dask/dask/pull/7325/commits/012427342c0bc452675adbde6ebd47dfa18107b4

Reply:
Thanks Peter!

The Dask PR removed the following rounding step after the `_percentiles` call:
```python
    if interpolation == "linear" and np.issubdtype(data.dtype, np.integer):
        vals = np.round(vals).astype(data.dtype)
```

Do we still need this if the data is cupy?

Reply:
I'm not sure about it, what I can say is that `dask_cudf` tests that were failing because of that in tonight's tests (see https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-gpu-test/CUDA=10.1,GPU_LABEL=gpu,OS=ubuntu16.04,PYTHON=3.7/1184/#showFailuresLink) pass for me locally. I'm happy to add that back if someone is confident on the correct usage here.

Reply:
Got it - Makes perfect sense to use CI as a guide.  My worry is that we may run into other problems if we were previously ensuring that divisions were the same dtype as the data, and now it will always be float.

Maybe, while we are already making a change, we can fix the origial issue for cupy data, and set the 0 and 100 percentiles to the actual min/max values?

Reply:
To follow up on my previous comment.  It may be best to do this after the `_percentiles` call:

```python
if is_cupy_type(data) and interpolation == "linear" and np.issubdtype(data.dtype, np.integer):
    vals = np.round(vals).astype(data.dtype)
    if qs[0] == 0:
        # Ensure the 0th quantile is the minimum value of the data
        vals[0] = data.min()
```

Reply:
Thanks @rjzamora for the suggestion, I can confirm locally `dask_cudf` tests also pass with the change above, so I think it's probably good to do it. I pushed https://github.com/dask/dask/pull/7325/commits/3a9fdf5d0cdd8472b23ca86201099e98d110a35e with the changes you suggested now.